sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -83,17 +83,21 @@ def __init__(
         self.id = id
         self.container_link = "{}/colls/{}".format(database_link, self.id)
         self.client_connection = client_connection
-        self._properties = properties
         self._is_system_key: Optional[bool] = None
         self._scripts: Optional[ScriptsProxy] = None
+        if properties:
+            self.client_connection.container_properties_cache[self.container_link] = {
+                "_self": properties.get("_self", None), "_rid": properties.get("_rid", None),
```

Comment:
nit: The duplication in the logic here and in line 182 and in the other file on which properties to extract can lead to in the future, to someone updating 1 place only instead of all 4. Would it be better to have a shared function that creates the cached entry based on the input and share that function across? So in the future, if you want to add more properties to the cache, you touch 1 place and not 4.

Reply:
Very good point, I made the change to make a single method to populate the properties cache.